src/api_server/src/parsed_request.rs

```diff
@@ -662,23 +662,45 @@ pub(crate) mod tests {
 
     #[test]
     fn test_try_from_put_machine_config() {
-        let (mut sender, receiver) = UnixStream::pair().unwrap();
-        let mut connection = HttpConnection::new(receiver);
-        sender
-            .write_all(
-                b"PUT /machine-config HTTP/1.1\r\n\
-                Content-Type: application/json\r\n\
-                Content-Length: 80\r\n\r\n{ \
-                \"vcpu_count\": 0, \
-                \"mem_size_mib\": 0, \
-                \"ht_enabled\": true, \
-                \"cpu_template\": \"C3\" \
-            }",
-            )
-            .unwrap();
-        assert!(connection.try_read().is_ok());
-        let req = connection.pop_parsed_request().unwrap();
-        assert!(ParsedRequest::try_from_request(&req).is_ok());
+        #[cfg(target_arch = "x86_64")]
+        {
+            let (mut sender, receiver) = UnixStream::pair().unwrap();
+            let mut connection = HttpConnection::new(receiver);
+            sender
+                .write_all(
+                    b"PUT /machine-config HTTP/1.1\r\n\
+                    Content-Type: application/json\r\n\
+                    Content-Length: 80\r\n\r\n{ \
+                        \"vcpu_count\": 0, \
+                        \"mem_size_mib\": 0, \
+                        \"ht_enabled\": true, \
+                        \"cpu_template\": \"C3\" \
+                    }",
+                )
+                .unwrap();
+            assert!(connection.try_read().is_ok());
+            let req = connection.pop_parsed_request().unwrap();
+            assert!(ParsedRequest::try_from_request(&req).is_ok());
+        }
+        #[cfg(target_arch = "aarch64")]
+        {
+            let (mut sender, receiver) = UnixStream::pair().unwrap();
+            let mut connection = HttpConnection::new(receiver);
+            sender
+                .write_all(
+                    b"PUT /machine-config HTTP/1.1\r\n\
+                    Content-Type: application/json\r\n\
+                    Content-Length: 58\r\n\r\n{ \
+                        \"vcpu_count\": 0, \
+                        \"mem_size_mib\": 0, \
+                        \"ht_enabled\": true \
+                    }",
+                )
+                .unwrap();
+            assert!(connection.try_read().is_ok());
+            let req = connection.pop_parsed_request().unwrap();
+            assert!(ParsedRequest::try_from_request(&req).is_ok());
+        }
```

Comment:
This 2 pieces of code are a bit redundant. Can we just build the string differently depending on the architecture ?

Reply:
Yes, that's a valid point

Reply:
Fixed it

Reply:
Another option would be something like 

```
                [
                    "PUT /machine-config HTTP/1.1\r\n\
                    Content-Type: application/json\r\n",
                    #[cfg(target_arch = "x86_64")]
                    "Content-Length: 58\r\n\r\n{ ",
                    #[cfg(target_arch = "aarch64")]
                    "Content-Length: 80\r\n\r\n{ ",
                    "\"vcpu_count\": 0, \
                    \"mem_size_mib\": 0, \
                    \"ht_enabled\": true, ",
                    #[cfg(target_arch = "x86_64")]
                    "\"cpu_template\": \"C3\" \
                    }",
                ]
                .concat()
                .as_bytes(),
```

Sine the strings are also quite redundant. Not sure which version is more readable though.

Reply:
I think that the current implementation is more readable. Having the attributes inside the request payload is a bit confusing, don't you think?